### DISCUSSION TITLE
enable zsh completion

### DISCUSSION THREAD

**Discussion by riolly** [geldata/gel#4147] [Q&A]
Can somebody explain to me what this mean?

```
To enable zsh completion, add:
fpath+=~/.zfunc
to your ~/.zshrc before compinit command.
```

Sorry for my silly question.

**Comment by tailhook** [comment]
1. Open `~/.zshrc` file (which is `.zshrc` in your `$HOME`) with your favorite editor.
2. Look if there are any `fpath=` or `fpath+=` lines. If there are, look if there are mentions of `~/.zfunc`. If there is, you're good.
3. If there are no `~/.zfunc` in any `fpath` lines, add mentioned one `fpath+=~/.zfunc` somewhere around them.
4. If there are no `fpath..` lines, insert the mentioned one `fpath+=~/.zfunc` somewhere before the line that mentions `compinit`. If there is no latter, you probably need to add one (but I think this would mean that no completion works for your zsh, refer to documentation of zsh to find out).

**Reply by riolly** [reply]
What makes me confused is there is no `compinit` in my .zsh.
I just added the `fpath+=~/.zfunc` to my .zshrc without adding `compinit` and the edgedb autocompletion work

**Reply by tailhook** [reply]
This is what `man zshcomsys` says:
> This  section  describes  the  use of compinit to initialize completion for the current session when called directly; if you have run compinstall it will be called automatically from your .zshrc.
>
> To initialize the system, the function compinit should be in a directory mentioned in the fpath parameter, and should be  autoloaded  (`autoload  -U compinit`  is  recommended), and then run simply as `compinit`.

I'm not sure I clearly understand what "will be called automaticallly from your .zshrc" means.

But there is also:
> This will define a few utility functions, arrange for all the necessary shell functions to be autoloaded, and will then re-define all widgets that do completion to use the new system.

So maybe you're using "old system"?

But anyways, if it works for you that's great.
